march_rqt_gait_selection/src/march_rqt_gait_selection/gait_selection_view.py

```diff
@@ -157,6 +165,17 @@ def update_version_colors(self):
             subgait_name = subgait_label.text()
             if subgait_name != 'Unused':
                 try:
+                    if 'parametric' == str(subgait_menu.currentText()):
+                        versions = self.available_gaits[gait_name][subgait_name]
+                        if self._parametric_pop_up.show_pop_up(versions):
+                            new_version = self.get_parametric_version()
+                            subgait_label.setStyleSheet('color:{color}'.format(color=self._colors['warning']))
+                            subgait_menu.addItem(new_version)
+                            subgait_menu.setCurrentIndex(subgait_menu.count() - 1)
+                        else:
+                            # parametric pop up window unsuccessful stopped, reset version to default
+                            current_version_index = versions.index(self.version_map[gait_name][subgait_name])
+                            subgait_menu.setCurrentIndex(current_version_index - 1)
```

Comment:
When the `current_version_index` is 0 you get an out of bounds. You can try by choosing parametric on a subgait with only 1 subgait version and then closing it.